internal/validitywindow/interfaces.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validitywindow
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/internal/emap"
+)
+
+type ExecutionBlock[Container emap.Item] interface {
+	Parent() ids.ID
+	Timestamp() int64
+	Height() uint64
+	Txs() []Container
+	ContainsTx(ids.ID) bool
+}
+
+type ChainIndex[Container emap.Item] interface {
+	GetExecutionBlock(ctx context.Context, blkID ids.ID) (ExecutionBlock[Container], error)
+}
+
+type TimeValidityWindow[Container emap.Item] interface {
+	Accept(blk ExecutionBlock[Container])
+	IsRepeat(
+		ctx context.Context,
+		parentBlk ExecutionBlock[Container],
+		txs []Container,
+		oldestAllowed int64,
+	) (set.Bits, error)
+	VerifyExpiryReplayProtection(
+		ctx context.Context,
+		blk ExecutionBlock[Container],
+		oldestAllowed int64,
+	) error
+}
+
+type Syncer[Container emap.Item] interface {
+	Accept(ctx context.Context, blk ExecutionBlock[Container]) (bool, error)
+}
```

Comment:
Can we move these into the relevant packages and remove the interfaces where we only define a single implementation ie. `Syncer` and `TimeValidityWindow` ? Open to pushback if there's a specific reason for keeping them as interfaces if you want to mock them elsewhere, but don't see that in this PR.

Reply:
ok, updated. could you take another look ?